src/main/java/rx/internal/operators/OperatorToObservableList.java

```diff
@@ -101,6 +107,9 @@ public void onNext(T value) {
             }
 
         };
+        o.add(result);
```

Comment:
Because if the result ```Subscriber``` receives its ```Producer``` via ```setProducer```, the default behavior is to call ```setProducer``` on ```op``` set by the constructor and thus can overwrite our ```SingleDelayedProducer```.